mlflow/transformers.py

```diff
@@ -0,0 +1,793 @@
+    :param registered_model_name: This argument may change or be removed in a
+                                  future release without warning. If given, create a model
+                                  version under ``registered_model_name``, also creating a
+                                  registered model if one with the given name does not exist.
+    :param signature: :py:class:`Model Signature <mlflow.models.ModelSignature>` describes model
+                      input and output :py:class:`Schema <mlflow.types.Schema>`. The model
+                      signature can be :py:func:`inferred <mlflow.models.infer_signature>`
+                      from datasets with valid model input (e.g. the training dataset with target
+                      column omitted) and valid model output (e.g. model predictions generated on
+                      the training dataset), for example:
+
+                      .. code-block:: python
+
+                        # TODO: fill out example once pyfunc support for dtypes is done
+
+    :param input_example: Input example provides one or several instances of valid
+                          model input. The example can be used as a hint of what data to feed the
+                          model. The given example will be converted to a ``Pandas DataFrame`` and
+                          then serialized to json using the ``Pandas`` split-oriented format.
+                          Bytes are base64-encoded.
+    :param await_registration_for: Number of seconds to wait for the model version
+                                   to finish being created and is in ``READY`` status.
+                                   By default, the function waits for five minutes.
+                                   Specify 0 or None to skip waiting.
+    :param pip_requirements: {{ pip_requirements }}
+    :param extra_pip_requirements: {{ extra_pip_requirements }}
+    :param metadata: Custom metadata dictionary passed to the model and stored in the MLmodel file.
+
+                     .. Note:: Experimental: This parameter may change or be removed in a future
+                                             release without warning.
+    :param kwargs: Additional arguments for :py:class:`mlflow.models.model.Model`
+    """
+    return Model.log(
+        artifact_path=artifact_path,
+        flavor=mlflow.transformers,
+        registered_model_name=registered_model_name,
+        await_registration_for=await_registration_for,
+        metadata=metadata,
+        transformers_model=transformers_model,
+        processor=processor,
+        task=task,
+        model_card=model_card,
+        conda_env=conda_env,
+        code_paths=code_paths,
+        signature=signature,
+        input_example=input_example,
+        pip_requirements=pip_requirements,
+        extra_pip_requirements=extra_pip_requirements,
+        **kwargs,
+    )
+
+
+def load_model(model_uri: str, dst_path: str = None, return_type="pipeline", **kwargs):
```

Comment:
Ditto here with `return_type` (https://github.com/mlflow/mlflow/pull/8086/files#r1147103069). I thought from the design review that we were always just going to accept pipelines as input / output for log_model and load_model. Can we no longer do this due to incompatibilities with Processors and pipelines in recent HF versions?

Reply:
If Processor is the limitation, I'm fine with not supporting use cases that require Processor initially, since we won't support pyfunc for image models, models with multi-modal inputs, etc. anyway. I think pipeline is easier.

Reply:
It's slightly more nuanced for the use cases. Based on a few interactions with users of the library and validating if all available processing objects are compatible with Pipelines (they most certainly are not - many cases the output data structures from the 'wrong' FeatureExtractor render the Pipeline unusable due to I/O formatting incompatibilities internal to the Pipeline), if we don't support at least the loading of components, users will be unable to use the flavor for a significant portion of use cases.

Reply:
Still a bit fuzzy here...what does 'wrong' FeatureExtractor mean and does any of this have an impact on LLM workflows?

Reply:
The incorrect feature extractor was 100% a result of my inference logic (take a model, modify some configurable aspects of it, then infer the extractor type from the base class type's name_or_path attribute could, for some LLM's, pull a feature extractor that is incompatible with the model (the Exception raised by transformers was high level and didn't expose the full stack trace, unfortunately); basically it was inferring a FeatureExtractor for a different LLM use case (i.e., the original problem is a question-answer task, but the inference pulled in the FeatureExtractor for text-classification, which throws). 
By removing the inference logic completely, confirmed that this problem goes away completely.